src/components/TrackingProvider/TrackingProvider.js

```diff
@@ -16,6 +16,7 @@
 import React, {PureComponent} from 'react';
 import PropTypes from 'prop-types';
 import TrackingContext from '../../context/TrackingContext';
+const merge = require('merge-deep');
```

Comment:
We're trying to keep this module dependency free that can run on both server and client. Probably thing to do here is to write a small deep merge function.

Reply:
Got it!
Ideally, a good deep merge function typically covers cases such that Object, Arrays and other primitive and non-primitive data types and should merge recursively. It should also be able to handle changes in data type where obj 1 sends the data of type 1 and obj 2 sends data of type 2 and possibly handle many other cases. 

The purpose we're trying to achieve is only with Object & String types and only extends to 3 levels deep right now. 

So the question is, What all cases should this merge module cover?